paddle/fluid/platform/mkldnn_helper.h

```diff
@@ -457,13 +466,14 @@ inline std::string CreateKey(const platform::MKLDNNDeviceContext& dev_ctx,
   key.reserve(64);
   using expand_type = int[];
   expand_type{0, (AppendKey(&key, std::forward<ArgTypes>(args)), 0)...};
-  key += dev_ctx.GetKeySuffix();
+  key += paddle::platform::MKLDNNDeviceContext::tls().get_key_suffix();
   return key;
 }
 
 inline std::string ExtendKeyWithThreadInfoIfNeeded(
     const platform::MKLDNNDeviceContext& dev_ctx, const std::string& key) {
-  return ((dev_ctx.IsThreadIdUsedInKey() == true) &&
+  return ((paddle::platform::MKLDNNDeviceContext::tls().is_tid_used_in_key() ==
```

Comment:
Could here exist some comments?
I also have a question, following
```
platform::MKLDNNDeviceContext::tls().get_cur_mkldnn_session_id() ==
           platform::MKLDNNDeviceContextThreadLocals::kMKLDNNSessionID_Default
``` 
means it is not limited_cache mode (cache clearing mode)
together means
```
if [is_tid_used_in_key (which is default true and for first thread, false) and unlimited cache capacity mode]:
    thread_id will be added into the key
```
but if it is cache clearing mode, thread id will not be added into the key?  Will there be some risks? If many threads execute same executor and use limited cache, why thread_id not needed in the key ?

Reply:
That is a valid concern. I remember that ClearCache mode was used only for single threaded execution, but when I look at code now , I do not see this limitation anywhere, so perhaps we should investgate this clearing cache functionality and if keys should be updated for it as well. I'm not going to look to that in this PR, so please file an issue in our internal backtracker.

Reply:
Ok I will make a jira task for this case.